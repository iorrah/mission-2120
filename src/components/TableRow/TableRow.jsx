import React from "react";
import { date, currency } from "../../utils/formatter.js";

const TableRow = props => (
  <tr className="main__table-row" onClick={props.handleClick}>
    <td>
      <input
        type="checkbox"
        className="main__checkbox"
        onClick={props.handleClickCheckbox}
      />
    </td>

    <td>{props.consumerId}</td>

    <td>
      <img
        src={`img/logo-${props.consumerId}.png`}
        className="main__row-logo"
        alt="Company Logo"
      />
    </td>

    <td>{props.consumerName}</td>
    <td>€{currency(props.consumerBudget)}</td>
    <td>€{currency(props.consumerBudgetSpent)}</td>
    <td>{date(props.consumerDateFirstPurchase)}</td>

    <td>
      <img
        src="img/icon-8.png"
        className="main__row-action"
        alt="Further actions"
      />
    </td>
  </tr>
);

export default TableRow;
