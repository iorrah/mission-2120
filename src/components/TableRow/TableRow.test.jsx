import React from "react";
import { render } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import TableRow from "./TableRow";

const props = {
  handleClick: jest.fn(),
  handleClickCheckbox: jest.fn(),
  consumerId: 1,
  consumerName: "Apple",
  consumerBudget: 123456.789,
  consumerBudgetSpent: 987.6543,
  consumerDateFirstPurchase: "2120-03-18"
};

describe("TableRow", () => {
  it("should shallow correctly", () => {
    expect(shallow(<TableRow {...props} />)).toMatchSnapshot();
  });

  it("should mount correctly", () => {
    expect(mount(<TableRow {...props} />)).toMatchSnapshot();
  });

  it("should render correctly", () => {
    expect(render(<TableRow {...props} />)).toMatchSnapshot();
  });

  it("should trigger parent click handler", () => {
    const mockCallBack = jest.fn();
    const wrapper = shallow(<TableRow {...props} handleClick={mockCallBack} />);
    wrapper.find("tr").simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it("renders the `.main__checkbox` input", () => {
    const wrapper = shallow(<TableRow {...props} />);
    expect(wrapper.find(".main__checkbox").exists()).toBeTruthy();
  });

  it("renders the `.main__row-logo` company logo", () => {
    const wrapper = shallow(<TableRow {...props} />);
    expect(wrapper.find(".main__row-logo").exists()).toBeTruthy();
  });

  it("renders the `.main__row-action` icon", () => {
    const wrapper = shallow(<TableRow {...props} />);
    expect(wrapper.find(".main__row-action").exists()).toBeTruthy();
  });

  it("renders 8 columns `<td>`", () => {
    const wrapper = shallow(<TableRow {...props} />);
    expect(wrapper.find("td").length).toBe(8);
  });
});
