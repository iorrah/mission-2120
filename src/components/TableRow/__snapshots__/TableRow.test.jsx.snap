// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow should mount correctly 1`] = `ReactWrapper {}`;

exports[`TableRow should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr
        class="main__table-row"
      >
        <td>
          <input
            class="main__checkbox"
            type="checkbox"
          />
        </td>
        <td>
          1
        </td>
        <td>
          <img
            alt="Company Logo"
            class="main__row-logo"
            src="img/logo-1.png"
          />
        </td>
        <td>
          Apple
        </td>
        <td>
          €
          123.456,79
        </td>
        <td>
          €
          987,65
        </td>
        <td>
          18/03/2120
        </td>
        <td>
          <img
            alt="Further actions"
            class="main__row-action"
            src="img/icon-8.png"
          />
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr
      class="main__table-row"
    >
      <td>
        <input
          class="main__checkbox"
          type="checkbox"
        />
      </td>
      <td>
        1
      </td>
      <td>
        <img
          alt="Company Logo"
          class="main__row-logo"
          src="img/logo-1.png"
        />
      </td>
      <td>
        Apple
      </td>
      <td>
        €
        123.456,79
      </td>
      <td>
        €
        987,65
      </td>
      <td>
        18/03/2120
      </td>
      <td>
        <img
          alt="Further actions"
          class="main__row-action"
          src="img/icon-8.png"
        />
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TableRow should shallow correctly 1`] = `ShallowWrapper {}`;
